### ISSUE TITLE
docker npm attempt registry request try #1

### ISSUE THREAD

**Issue by jasonkuhrt** [npm/npm#15862]
Whenever we try `npm install` within a docker container lately we see following:

<img width="1680" alt="208e1746-f38b-11e6-8970-c4ba02ebc1d3" src="https://cloud.githubusercontent.com/assets/284476/23307115/5486ca26-fa74-11e6-9814-2be76935c780.png">

This sometimes happens for many minutes before finally finishing install, or sometimes seemingly indefinitely. I'll try to provide more details in this thread the next time it happens.

Is this is a known issue I've seen similar issues to this reported in 2016 but most seem closed. Some talk about changing the values of `/etc/hosts` on the docker image but that seems drastic.

**Comment by legodude17** [comment]
It is probably an issue with your network.

**Comment by zkat** [comment]
@jasonkuhrt I don't see anything wrong here. This is just npm's debugging. The `attempt` thing is just `npm-registry-client` doing its first attempt, and not a request failure. Try passing `--loglevel=warn` to npm, and it should be a little quieter.

Based on this, my best guess is you're just having network issues (which is more of a node and docker thing), which is sort of out of our hands. :\

**Comment by jasonkuhrt** [comment]
Hm, but interestingly I've seen this predictably on different networks (work, cafe, home) inside `docker build ...`, but generally never outside them (I think).

**Comment by legodude17** [comment]
That might be a problem with `docker` not forwarding `npm`s requests?

**Comment by jasonkuhrt** [comment]
Here's another recent example. Note how npm install fails within docker but succeeds outside it (here `dc` is `docker-compose`).

```
❯ dc exec ui-website npm install
npm info it worked if it ends with ok
npm info using npm@3.10.10
npm info using node@v6.9.5
npm info retry fetch attempt 1 at 7:04:19 AM
npm info attempt registry request try #1 at 7:04:19 AM
npm http fetch GET https://registry.npmjs.org/fsevents/-/fsevents-1.0.17.tgz
npm info retry fetch attempt 1 at 7:04:19 AM
npm info attempt registry request try #1 at 7:04:19 AM
npm http fetch GET http://pdx-artifacts.pdx.vm.datanerd.us:8081/artifactory/api/npm/npmjs-remote/concat-stream/-/concat-stream-1.6.0.tgz
npm info retry fetch attempt 1 at 7:04:19 AM
npm info attempt registry request try #1 at 7:04:19 AM
npm http fetch GET http://pdx-artifacts.pdx.vm.datanerd.us:8081/artifactory/api/npm/npmjs-remote/yakaa/-/yakaa-1.0.1.tgz
npm info retry fetch attempt 1 at 7:04:19 AM
npm info attempt registry request try #1 at 7:04:19 AM
npm http fetch GET http://pdx-artifacts.pdx.vm.datanerd.us:8081/artifactory/api/npm/npmjs-remote/https-proxy-agent/-/https-proxy-agent-0.3.6.tgz
npm http fetch 200 https://registry.npmjs.org/fsevents/-/fsevents-1.0.17.tgz

~/projects/ssense/ws-website/services/ui-website master 1m 13s
❯ npm install
npm WARN deprecated inversify-dts@2.2.0: Type definitions are included in the inversify npm releases since the release of inversify@2.0.0. Learn more at https://github.com/inversify/InversifyJS#installation
npm WARN deprecated ghooks@1.3.2: Use npmjs.com/husky instead, see https://github.com/gtramontina/ghooks/issues/166
- mime@1.2.11 node_modules/url-loader/node_modules/mime
- babel-polyfill@6.22.0 node_modules/babel-polyfill
ui-website@1.6.6 /Users/jason.kuhrt/projects/ssense/ws-website/services/ui-website
├─┬ autoprefixer@6.7.6
│ ├─┬ browserslist@1.7.6
│ │ └── electron-to-chromium@1.2.5
│ ├── caniuse-db@1.0.30000632
│ └── postcss@5.2.15
├─┬ babel-core@6.23.1
│ └── babylon@6.16.1
├── babel-loader@6.4.0
├─┬ css-loader@0.23.1
│ └─┬ cssnano@3.10.0
│   ├── postcss-discard-duplicates@2.1.0
│   └── postcss-merge-rules@2.1.2
├─┬ eslint@3.17.0
│ ├── globals@9.16.0
│ ├── ignore@3.2.4
│ └─┬ table@3.8.3
│   └── ajv@4.11.4
├── eslint-config-airbnb-base@11.1.1
├─┬ eslint-plugin-vue@1.0.0
│ └─┬ eslint-plugin-react@6.10.0
│   ├─┬ array.prototype.find@2.0.3
│   │ └─┬ es-abstract@1.7.0
│   │   └── is-regex@1.0.4
│   └── object.assign@4.0.4
├─┬ jest@18.1.0
│ └─┬ jest-cli@18.1.0
│   ├─┬ istanbul-api@1.1.1
│   │ └── async@2.1.5
│   └─┬ jest-environment-jsdom@18.1.0
│     └─┬ jsdom@9.11.0
│       ├── webidl-conversions@4.0.1
│       └── whatwg-url@4.5.0
├─┬ sass-loader@4.1.1
│ └── async@2.1.5
├─┬ style-loader@0.13.2
│ └── loader-utils@1.0.3
├─┬ tslint@3.15.1
│ └── resolve@1.3.2
├─┬ typings@1.5.0
│ └─┬ typings-core@1.6.1
│   └── typescript@2.2.1
├─┬ url-loader@0.5.8
│ └── loader-utils@1.0.3
├── vue-hot-reload-api@2.0.11
├─┬ vue-html-loader@1.2.4
│ ├─┬ es6-templates@0.2.3
│ │ └── recast@0.11.22
│ ├─┬ html-minifier@2.1.7
│ │ └── clean-css@3.4.25
│ └── loader-utils@1.0.3
├─┬ vue-loader@10.0.2
│ ├── js-beautify@1.6.11
│ ├── postcss-selector-parser@2.2.3
│ └── vue-template-es2015-compiler@1.5.1
├── vuelidate@0.2.0
├─┬ webpack-bundle-analyzer@1.5.4
│ ├── filesize@3.5.5
│ └── opener@1.4.3
├── webpack-dev-middleware@1.10.1
└─┬ webpack-hot-middleware@2.17.1
  └── ansi-html@0.0.7
```

**Comment by legodude17** [comment]
@jasonkuhrt There aren't any errors in the output you posted. What is the problem?

**Comment by jasonkuhrt** [comment]
@legodude17 That `dc exec ui-website npm install` would seemingly resolve (I stopped at 1m 37s but had it run previously for much longer).

Interestingly I had been connected to a VPN before. I restarted my computer, tried the command again without success. Then I decided to destroy the containers and rebuild them from the image. This worked... Maybe the issue is on the docker side rather than npm (seems that way now).
